recipes/restinio/all/conanfile.py

```diff
@@ -0,0 +1,82 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class RestinioConan(ConanFile):
+    name = "restinio"
+    license = "BSD-3-CLAUSE"
+    homepage = "https://github.com/Stiffstream/restinio"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "RESTinio is a header-only C++14 library that gives you an embedded HTTP/Websocket server."
+    topics = ("http-server", "websockets", "rest", "tls-support")
+    exports_sources = ["CMakeLists.txt"]
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"asio": ["boost", "standalone"], "with_openssl": [True, False], "with_zlib": [True, False]}
+    default_options = {"asio": "standalone", "with_openssl": False, "with_zlib": False}
+    generators = "cmake"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        tools.check_min_cppstd(self, "14")
```

Comment:
That's what I want, if the default value is lower than the required one then the user should set the value manually. The default value doesn't matters, the only value that matter is the one actually used for the build.

At least that how i thought it should works.

Reply:
But then the check for C++14 pass and the build fail because the compiler doesn't support C++14 (https://github.com/conan-io/conan-center-index/pull/1635#issuecomment-630425159). Any idea what's happening here ?

Reply:
Restinio forces C++14 by compiler flags. Even the if a compiler does not set c++14 as default, restinio will force it, so it should work most c++14 compatible compilers. Anyway, this is the current way for all recipes in CCI.

Reply:
Also, you need to check the supported compilers, otherwise, this recipe will break when running old versions. You can use this recipe as reference: https://github.com/conan-io/conan-center-index/blob/e1f7cdf1a7ce35de2c2c9febd882c6b5ffa66d8e/recipes/frozen/all/conanfile.py#L17

Reply:
Thanks for your help (and @prince-chrismc too) ! Sorry to have that many failed builds, i know it's not supposed to be a CI.

Reply:
Price of learning, I appreciate this being migrated so thank you very much :+1: